clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+        .containsExactlyInAnyOrder(
+            "service frontend", "deployment frontend", "service backendlb", "replicaSet backend");
+
+    Map<String, KubernetesV2Cluster> clusterLookup =
+        clusters.stream().collect(toImmutableMap(KubernetesV2Cluster::getName, c -> c));
+
+    assertFrontendServiceCluster(softly, clusterLookup.get("service frontend"));
+    assertFrontendCluster(softly, clusterLookup.get("deployment frontend"), false);
+    assertBackendServiceCluster(softly, clusterLookup.get("service backendlb"));
+    assertBackendCluster(softly, clusterLookup.get("replicaSet backend"), false);
+  }
+
+  @Test
+  void getClustersForApplication(SoftAssertions softly) {
+    Map<String, Set<KubernetesV2Cluster>> results = clusterProvider.getClusterDetails("backendapp");
+    assertThat(results).hasSize(1);
+    assertThat(results).containsKey(ACCOUNT_NAME);
+
+    Set<KubernetesV2Cluster> clusters = results.get(ACCOUNT_NAME);
+    assertThat(clusters).hasSize(2);
+
+    assertThat(clusters)
+        .extracting(KubernetesV2Cluster::getName)
+        .containsExactlyInAnyOrder("service backendlb", "replicaSet backend");
+
+    Map<String, KubernetesV2Cluster> clusterLookup =
+        clusters.stream().collect(toImmutableMap(KubernetesV2Cluster::getName, c -> c));
+
+    assertBackendServiceCluster(softly, clusterLookup.get("service backendlb"));
+    assertBackendCluster(softly, clusterLookup.get("replicaSet backend"), false);
+  }
+
+  @Test
+  void getClustersForApplicationAndAccount(SoftAssertions softly) {
+    Set<KubernetesV2Cluster> clusters = clusterProvider.getClusters("backendapp", ACCOUNT_NAME);
+    assertThat(clusters).hasSize(2);
+
+    assertThat(clusters)
+        .extracting(KubernetesV2Cluster::getName)
+        .containsExactlyInAnyOrder("service backendlb", "replicaSet backend");
+
+    Map<String, KubernetesV2Cluster> clusterLookup =
+        clusters.stream().collect(toImmutableMap(KubernetesV2Cluster::getName, c -> c));
+
+    assertBackendServiceCluster(softly, clusterLookup.get("service backendlb"));
+    assertBackendCluster(softly, clusterLookup.get("replicaSet backend"), false);
+  }
+
+  @Test
+  void getClustersForApplicationAndWrongAccount(SoftAssertions softly) {
+    Set<KubernetesV2Cluster> clusters = clusterProvider.getClusters("backendapp", "non-existent");
+    assertThat(clusters).hasSize(0);
+  }
+
+  @Test
+  void getSingleCluster(SoftAssertions softly) {
+    KubernetesV2Cluster cluster =
+        clusterProvider.getCluster("frontendapp", ACCOUNT_NAME, "deployment frontend");
+    assertThat(cluster).isNotNull();
+    assertFrontendCluster(softly, cluster, false);
+  }
+
+  @Test
+  void getSingleClusterWithDetails(SoftAssertions softly) {
+    KubernetesV2Cluster cluster =
+        clusterProvider.getCluster("frontendapp", ACCOUNT_NAME, "deployment frontend", true);
+    assertThat(cluster).isNotNull();
+    assertFrontendCluster(softly, cluster, false);
```

Comment:
Why is this the same assertion as the non-details return on line 245? Is the `summary` bool not actually relevant to whether or not details are included?

Reply:
Good catch, it looks like I didn't actually test this function when passing `includeDetails = false`.  This is the same as the test above because the test above doesn't pass the `includeDetails` flag at all (and the behavior is to default to having it `true`).

I've made two changes:
* Added a test when passing `includeDetails = false` so that now there are 3 (`true`, `false`, and omitted)
* I inverted the boolean in `assertFrontendCluster` so instead of passing `summary = false` when you want details you pass `includeDetails = true` so it matches the behavior of the function we're testing.  This hopefully makes it less confusing now that the function and the assertion have a flag with the same meaning.